### ISSUE TITLE
Docker studio breaks key ownership

### ISSUE THREAD

**Issue by themightychris** [habitat-sh/habitat#7737]
For a new user trying to get straight into studio, you can end up with commands crashing because your user can't access keys downloaded by studios.

Two issues surfaced here:

- `hab origin key export` crashing because it can't read everything it scans, even though it can read the key I'm asking for
- the studio leaves challenging permissions on the keys it downloads. Leaving .pubs globally readable and not writable at all seems like it would be an improvement

```console
$ ls -l ~/.hab
ls: cannot access '/home/chris/.hab': No such file or directory

$ hab --version
hab 1.6.39/20200527162613

$ hab setup

Habitat CLI Setup
=================
Connect to an on-premises Builder instance? No 
Set up a default origin? Yes
Default origin name: chris
Create an origin key for `chris'? Yes
» Generating origin key for chris
★ Generated origin key pair chris-20200605013300.
Set up a default Builder personal access token? No 
Set up a default Habitat Supervisor control gateway secret? No

$ ls -l ~/.hab/cache/keys/
total 8
-r-------- 1 chris chris  76 Jun  4 21:33 chris-20200605013300.pub
-r-------- 1 chris chris 120 Jun  4 21:33 chris-20200605013300.sig.key

$ hab origin key export --type secret chris
SIG-SEC-1
chris-20200605013411

ltGEJs4evpscXXYBjwD7fuv6tAESasr4Ku0cc1Lv9GIQBTinGJn4Z8wFoyCYWBM4LvvqURHCeQXoto9gNFzE0A==chris-desktop:~

$ HAB_DOCKER_OPTS="-d" hab studio enter -D
3913d908eca269cc076f2b422723b9320010f38c5b3f4214d0d52b626f7ffc3f

$ ls -l ~/.hab/cache/keys/
total 16
-r-------- 1 chris chris  76 Jun  4 21:33 chris-20200605013300.pub
-r-------- 1 chris chris 120 Jun  4 21:33 chris-20200605013300.sig.key
-rw------- 1 root  root   75 Jun  4 21:33 core-20180119235000.pub
-rw------- 1 root  root   75 Jun  4 21:33 core-20200219002627.pub

$ hab origin key export --type secret chris
✗✗✗
✗✗✗ Permission denied (os error 13)
✗✗✗
```

**Comment by stevendanna** [comment]
Related: https://github.com/habitat-sh/habitat/issues/7269

**Comment by stevendanna** [comment]
For the second half of this bug, we'll likely want to use the AtomicWriter's with_permission feature for those key files. 0444 for public keys and 0400 for private keys makes sense to me.

**Comment by davidMcneil** [comment]
The second half of this issue (ie "the studio leaves challenging permissions on the keys it downloads") should be resolved by #7739. This should also help with the first issue (ie it is less likely to have files with permission issues). Let's close this issue in favor of tracking the first part of the work in #7269.

**Comment by themightychris** [comment]
Thanks @davidMcneil !
